src/lndk_offers.rs

```diff
@@ -190,6 +196,29 @@ impl OfferHandler {
     }
 }
 
+// pay_invoice tries to pay the provided invoice.
+pub(crate) async fn pay_invoice(
+    mut payer: impl InvoicePayer + std::marker::Send + 'static,
+    path: BlindedPath,
+    cltv_expiry_delta: u16,
+    fee_base_msat: u32,
+    payment_hash: [u8; 32],
+    msats: u64,
+) -> Result<(), OfferError<Secp256k1Error>> {
+    let resp = payer
+        .query_routes(path, cltv_expiry_delta, fee_base_msat, msats)
+        .await
+        .map_err(OfferError::RouteFailure)?;
+
+    // TODO: If route fails, loop over resp.routes array and try alternatives.
+    let _send_resp = payer
+        .send_to_route(payment_hash, resp.routes[0].clone())
```

Comment:
Note that this API returning without error doesn't mean the payment was actually successful - it just means that the payment was dispatched to the network without error. You still need to call `trackpayment` to get the final outcome. I think there are some synchronous APIs available if you want this to block on payment success/failure, but I'm not sure they'll have blinded route support.